### PR TITLE
Documentation for density-based reward modeling

### LINKED ISSUE #721: Density-based reward modeling is missing documentation
This page about density-based reward modeling could have a short intro text to explain what it is about:

https://imitation.readthedocs.io/en/latest/algorithms/density.html

### PR DESCRIPTION
**Pull request by ernestum**
## Description

Add short intro about density-based reward modeling.
Fixes #721 

## Testing

Documentation only changes.

### CODE REVIEW

File: docs/algorithms/density.rst [hunk 1]
```diff
@@ -4,6 +4,13 @@
 Density-based reward modeling
 =============================
 
+Density-based reward modeling is an inverse reinforcement learning (IRL) technique
+that eliminates the need for explicit rewards by assigning higher rewards to states or state-action pairs
```
Comment (Rocamonde) [inline docs/algorithms/density.rst:8]: "explicit rewards" do you mean that the need that expert trajectories have rewards in them? I'm confused, isn't this how IRL normally works? (i.e. learning the rewards from trajectories without rewards)

File: docs/algorithms/density.rst [hunk 2]
```diff
@@ -4,6 +4,13 @@
...
+that occur more frequently in the expert's behavior.
+This variant leverages kernel density estimation to model the underlying distribution of expert demonstrations,
+enabling the generation of a reward model that captures the expert's preferences solely based on their observed
```
Comment (Rocamonde) [inline docs/algorithms/density.rst:11]: again, is this not just a description of imitation/reward learning?

File: docs/algorithms/density.rst (hunk 1 above)
Comment (ernestum) [reply docs/algorithms/density.rst:8]: No you are right I was the one being confused about the purpose of the density algorithm.

File: docs/algorithms/density.rst (hunk 2 above)
Comment (ernestum) [reply docs/algorithms/density.rst:11]: I tried to make it more concise.

**Review by Rocamonde — APPROVED** [review]

**Pull request merged**